trento/migration/sso-integration.md

```diff
@@ -0,0 +1,441 @@
+
+OAUTH2_SCOPES
+
+: OAUTH2 scopes, used to define the user values sent to the SP. It must be adjusted depending on IDP provider requirements (default value: `openid profile email`)
+
+OAUTH2_CALLBACK_URL
+
+: OAUTH2 callback url where the IDP is redirecting once the authentication is completed (default value: `https://#{TRENTO_WEB_ORIGIN}/auth/oauth2_callback`)
+
+
+## Using SAML
+
+Trento integrates with an IDP that uses the SAML protocol to authenticate users accessing the Trento web console. Trento will behave as a Service Provider (SP) in this case.
+
+Commonly, SAML protocol messages are signed with SSL. This is optional using Trento, and the signing is not required (even though it is recommended).
+If the IDP signs the messages, and expect signed messages back, certificates used by the SP (Trento in this case) must be provided to the IDP, the public certificate file in this case.
+
+To use an existing SAML IDP, follow the next instrunctions to met the specific requirements. You need:
+
+* Configure SAML IDP and user profiles
+* Start Trento to generate the certificates and download them
+* Provide the generated certificate to the IDP
+* Restart Trento
+
+### Configuring SAML IDP setup
+
+Configure the existing IDP with the next minimum options to be able to connect with Trento's Service Provider (SP).
+
+#### SAML user profile
+
+Users provided by the SAML installation must have some few mandatory attributes to login in Trento. All of them are mandatory, even though their name is configurable.
+The user profile must include attributes for: username, email, first name and last name.
+This attributes must be mapped for all users wanting to connect to Trento.
+
+By default, Trento expects the `username`, `email`, `firstName` and `lastName` attribute names. All these 4 attribute names are configurable using the next environment variables, following the same order: `SAML_USERNAME_ATTR_NAME`, `SAML_EMAIL_ATTR_NAME`, `SAML_FIRSTNAME_ATTR_NAME` and `SAML_LASTNAME_ATTR_NAME`.
+So for example, if the IDP user profile username is defined as `attr:username`, `SAML_USERNAME_ATTR_NAME=attr:username` should be used.
+
+#### SAML redirect URI
+
+Once the login is done succesfully, the IDP redirects the session back to Trento. This redirection is done to `https://trento.example.com/sso/sp/consume/saml`, so this URI must be set as valid in the IDP.
+
+### Get certificates from Trento
+
+Trento provides a certificates set created during the installation. When Trento is installed the first time (does not matter the installation mode) the certificates are created, and the public certificate file content is available in the <uri>http://localhost:4000/api/public_keys</uri> route. 
+
+Use the following command to download the certificate:
+
+```bash
+curl http://localhost:4000/api/public_keys
+```
+    
+Copy the content of the certificate from there, and provide it to the IDP. This way, the IDP will sign and verify the messages sent by both ends.
```

Comment:
What do you mean by "provide it to the IDP"? And "both ends" means Trento and IDP, right?

```suggestion
Copy the content of the certificate from there and provide it to the IDP. This way, the IDP will sign and verify the messages sent by both ends.
```

Reply:
No, Trento already have it (it was created by Trento), so you need to copy the content and provide to the IDP, and we cannot say how you do that, because this depends on the IDP. But all IDPs will have the option to use it

Reply:
I will rephrase to `Copy the content of the certificate from there and provide it to the IDP. This way, the IDP will sign its messages and verify the messages received from Trento.`